File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int curr_sum = 0;
        int max_sum = Integer.MIN_VALUE;

        for(int num:nums)
        {
            curr_sum+=num;

            if(curr_sum > max_sum)
            {
                max_sum = curr_sum;
            }

            if(curr_sum<0)
            {
                curr_sum = 0;
            }

            
        }

        return max_sum;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        totalSum = 0
        maxSum = -10001

        for num in nums:
            totalSum += num

            maxSum = max(maxSum,totalSum)

            if totalSum<0:
                totalSum = 0
        

        return maxSum
        